astronomer/providers/amazon/aws/example_dags/example_sagemaker.py

```diff
@@ -203,20 +214,82 @@ def set_up(role_arn: str) -> None:
-def delete_logs() -> None:
+def delete_logs(task_instance: "TaskInstance") -> None:
     """Delete the cloud watch log based on the log group name"""
+    import boto3
+    from botocore.exceptions import ClientError
+
     generated_logs = [
         "/aws/sagemaker/ProcessingJobs",
         "/aws/sagemaker/TrainingJobs",
         "/aws/sagemaker/TransformJobs",
     ]
+    sagemaker_credentails_xcom = task_instance.xcom_pull(
+        key="sagemaker_credentials", task_ids=["get_aws_sagemaker_session_details"]
+    )[0]
+    creds = {
+        "aws_access_key_id": sagemaker_credentails_xcom["AccessKeyId"],
+        "aws_secret_access_key": sagemaker_credentails_xcom["SecretAccessKey"],
+        "aws_session_token": sagemaker_credentails_xcom["SessionToken"],
+        "region_name": AWS_DEFAULT_REGION,
+    }
+    client = boto3.client("logs", **creds)
+    for group in generated_logs:
+        try:
+            if client.describe_log_streams(logGroupName=group)["logStreams"]:
+                client.delete_log_group(logGroupName=group)
+        except ClientError as e:
+            raise e
+
+
+def get_aws_sagemaker_session(task_instance: "TaskInstance") -> None:
+    """Get session details by using env variables credentials details"""
     import boto3
+    from botocore.exceptions import ClientError
 
-    client = boto3.client("logs")
-    for group in generated_logs:
-        if client.describe_log_streams(logGroupName=group)["logStreams"]:
-            client.delete_log_group(logGroupName=group)
+    client = boto3.client("sts", **AWS_SAGEMAKER_CREDS)
+    try:
+        response = client.get_session_token(DurationSeconds=10800)
+        task_instance.xcom_push(
+            key="sagemaker_credentials",
+            value=json.loads(json.dumps(response["Credentials"], cls=AirflowJsonEncoder)),
+        )
+    except ClientError as e:
+        raise e
+
+
+def setup_sagemaker_connection_details(task_instance: "TaskInstance") -> None:
+    """
+    Checks if airflow connection exists, if yes then deletes it.
+    Then, create a new aws_sagemaker_default connection.
+    """
+    creds_details = task_instance.xcom_pull(
+        key="sagemaker_credentials", task_ids=["get_aws_sagemaker_session_details"]
+    )[0]
+    print("creds_details ", creds_details)
+    print("creds_details type ", type(creds_details))
+    conn = Connection(
+        conn_id=SAGEMAKER_CONN_ID,
+        conn_type="aws",
+        login=creds_details["AccessKeyId"],
+        password=creds_details["SecretAccessKey"],
+        extra=json.dumps(
+            {"region_name": AWS_DEFAULT_REGION, "aws_session_token": creds_details["SessionToken"]}
+        ),
+    )  # create a sagemaker connection object
+
+    session = settings.Session()
+    connection = session.query(Connection).filter_by(conn_id=conn.conn_id).one_or_none()
+    if connection is None:
+        logging.info("Connection %s doesn't exist.", str(conn.conn_id))
+    else:
+        session.delete(connection)
```

Comment:
what if a user already has `aws_sagemaker_default` connection in their set up. If they run this DAG, then it will get deleted. Use some uncommon name like `aws_sagemaker_async_conn`

Reply:
Fixed the connection to `aws_sagemaker_async_conn`